frb_codegen/src/ir/ty_boxed.rs

```diff
@@ -45,11 +53,18 @@ impl IrTypeTrait for IrTypeBoxed {
         }
     }
 
-    fn rust_wire_type(&self) -> String {
-        self.inner.rust_wire_type()
+    fn rust_wire_type(&self, wasm: bool) -> String {
+        if wasm && !self.inner.is_js_value() {
+            format!("usize /* {} */", self.inner.rust_wire_type(wasm))
```

Comment:
This sounds a bit nontrivial. Would be great if there are some very brief design doc discussing some high-level implementation details :)

Reply:
Not much, this is because integers are the next best thing in the absence of pointers. (I'm wondering if isize might be better)

Reply:
Looking at wasm_bindgen docs reminded me that `Option<*mut T>` is not a valid JS type, but `Option<usize>` is, explaning this piece of code.